### PR TITLE
docs: updated readme for 2u/release contributions

### PR DESCRIPTION
**Pull request by wdrussell2015**
# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -30,6 +30,8 @@ The code in this repository is licensed under version 3 of the AGPL unless other
 How To Contribute
 -----------------
 
+Notice: Internal 2U contributions should be made against the `2u/release` branch.  Open source contributions should continue to be made against the `master` branch.
```
Comment (nedbat) [inline README.rst]: We had discussed `2u/main`.

Comment (wdrussell2015) [reply README.rst]: Updated, This PR had gone up before we spoke.

File: README.rst [hunk 2]
```diff
@@ -30,6 +30,8 @@ The code in this repository is licensed under version 3 of the AGPL unless other
 How To Contribute
 -----------------
 
+Notice: Internal 2U contributions should be made against the `2u/main` branch.  Open source contributions should continue to be made against the `master` branch.
```
Comment (nedbat) [inline README.rst]: .rst needs \`\` around monospaced text:
```suggestion
Notice: Internal 2U contributions should be made against the ``2u/main`` branch.  Open source contributions should continue to be made against the ``master`` branch.
```

File: README.rst [hunk 3]
```diff
@@ -30,6 +30,8 @@ The code in this repository is licensed under version 3 of the AGPL unless other
...
+
 Anyone merging to this repository is expected to `release and monitor their changes <https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories>`__; if you are not able to do this DO NOT MERGE, please coordinate with someone who can to ensure that the changes are released.
```
Comment (nedbat) [inline README.rst]: Should we change this paragraph?  Only the `2u/main` branch needs to be monitored, right?

**Review by nedbat — APPROVED** [review]

File: README.rst [hunk 4]
```diff
@@ -30,7 +30,9 @@ The code in this repository is licensed under version 3 of the AGPL unless other
 How To Contribute
 -----------------
 
-Anyone merging to this repository is expected to `release and monitor their changes <https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories>`__; if you are not able to do this DO NOT MERGE, please coordinate with someone who can to ensure that the changes are released.
+Notice: Internal 2U contributions should be made against the ``2u/main`` branch.  Open source contributions should continue to be made against the ``master`` branch.
+
+Anyone merging to the ``2u/main`` branch of this repository is expected to `release and monitor their changes <https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories>`__; if you are not able to do this DO NOT MERGE, please coordinate with someone who can to ensure that the changes are released.
```
Comment (julianajlk) [inline README.rst]: This link does not work, perhaps it should be - https://2u-internal.atlassian.net/wiki/spaces/RS/pages/7963261/How+to+contribute+to+our+repositories

File: README.rst [hunk 5]
```diff
@@ -30,7 +30,9 @@ The code in this repository is licensed under version 3 of the AGPL unless other
...
 
 Please also read `How To Contribute <https://github.com/openedx/edx-platform/blob/master/CONTRIBUTING.rst>`__. Even though it was written with ``edx-platform`` in mind, these guidelines should be followed for Open edX code in general.
```
Comment (julianajlk) [inline README.rst]: This link also appears broken and has moved to https://github.com/openedx/.github/blob/master/CONTRIBUTING.md which links to https://openedx.atlassian.net/wiki/spaces/COMM/pages/941457737/How+to+Start+Contributing+Code

File: docs/decisions/0008-master-branch-split.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+Both 2U and Openedx use ecommerce's master branch for releases. Occasionally, changes
```
Comment (nedbat) [inline docs/decisions/0008-master-branch-split.rst]:
```suggestion
Both 2U and the Open edX have been using ecommerce's master branch for releases. Occasionally, changes
```

File: docs/decisions/0008-master-branch-split.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+specific to 2U's business case are merged into code, which also influences the structure
+of the code that the community runs, even if the changes are not relevant or beneficial
+to the community at large.
+
+Additionally, 2U has internal compliance checks that can slow the release cycle of
+major changes to ecommerce (e.g. django 4.2). This puts Openedx at risk of releasing
```
Comment (nedbat) [inline docs/decisions/0008-master-branch-split.rst]:
```suggestion
major changes to ecommerce (for example, the upgrade to Django 4.2). This puts the Open edX community at risk of releasing
```

File: docs/decisions/0008-master-branch-split.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+with deprecated packages.
+
+Decision
+--------
+
+A new protected branch will be created named "2u/main", which 2U will continue to use
+as its own main branch, while leaving "master" branch to the community.
```
Comment (nedbat) [inline docs/decisions/0008-master-branch-split.rst]:
```suggestion
as its own main branch, while leaving the "master" branch for use by the community.
```

File: docs/decisions/0008-master-branch-split.rst [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+---------------------
+
+Repo fork
+~~~~~~~~~
+
+Why not fork the repo entirely? 2U and the community use a shared set of tools for
+working in development environments. While working from separate branches and repo
+forks are functionally equivalent, working in separate branches allows for
+minimal updates to shared tooling to support this change. One can simply checkout a
+branch name, without needing the tooling to be updated to know about non Openedx
```
Comment (nedbat) [inline docs/decisions/0008-master-branch-split.rst]:
```suggestion
branch name, without needing the tooling to be updated to know about non-openedx
```

**Review by nedbat — APPROVED** [review]
A few grammar tweaks. Good to go!

File: docs/decisions/0008-master-branch-split.rst (hunk 6 above)
Comment (christopappas) [reply docs/decisions/0008-master-branch-split.rst]: i see your change and raise you:`the Open edX community`

File: docs/decisions/0008-master-branch-split.rst [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+as its own main branch, while leaving "master" branch to the community.
+
+Consequences
+------------
+
+This allows the community to move forward with contributions on "master" without risk of
+breaking functionality for 2U or 2U committing changes to "master" that are irrelevant or
+otherwise inappropriate for the community's use case.
+
+Because protected branches in ecommerce require a user with Write access to the repository,
```
Comment (julianajlk) [inline docs/decisions/0008-master-branch-split.rst:34]: Could we add this information to the README? I am not sure it's clear master will now only rely on community review only. We could separate the "How to Contribute" section into Open Source and 2U contributions

**Review by julianajlk — APPROVED** [review]

File: README.rst (hunk 4 above)
Comment (christopappas) [reply README.rst]: So that is a 2U internal wiki link. I would actually be in favor of removing this link entirely if it is broken

Comment (nedbat) [reply README.rst]: It's ok to use a 2U wiki link, just mark it as `(2U-private link)`.

File: docs/decisions/0008-master-branch-split.rst (hunk 10 above)
Comment (christopappas) [reply docs/decisions/0008-master-branch-split.rst:34]: For what it's worth, how the community wants to interact with ecommerce is actually still an open question (most likely for Axim) that has been raised already. I'm not sure I could really add specifics around community review and master other than "2U wont be doing it" 🤔

File: README.rst (hunk 4 above)
Comment (christopappas) [reply README.rst]: I'm typically not a fan of internal links being external for the reason of "it's nobody's business how we function internally" (b/c it's more attack surface area). I'm guessing we do this in other places though, right?

File: README.rst (hunk 5 above)
Comment (christopappas) [reply README.rst]: fixed!

File: README.rst (hunk 4 above)
Comment (christopappas) [reply README.rst]: considering this a case of "disagree but commit." i've updated the link

**Pull request merged**